features/command_line_interface/show_progress.feature

```diff
@@ -0,0 +1,30 @@
+Feature: Show progress
+  In order to see the progress of the examiners
+  As a developer
+  I want to be able to selectively activate progress reporting
+
+  Scenario: --progress activates progress output on clean files
+    Given a directory called 'clean_files' containing some clean files
+    When I run reek -f progress clean_files
+    Then it succeeds
+    And it reports:
+      """
+      Inspecting 3 file(s):
+      ...
+
+      0 total warnings
+      """
+  Scenario: --progress activates progress output on smelly files
```

Comment:
Let's add a scenario where smelly and non-smelly files are mixed or update this to reflect this case.